### PR TITLE
Bug/sc 28476/error when trying to add captions for images

### PR DESCRIPTION
**Pull request by yonadavGit**
## Description
The "disambiguation" field of topic titles was not sent to the client. As a result, when editing a topic with disambiguations in its title, the inconsistency prevented the server from properly updating the topic object. This PR fixes the issue by making the client extract the disambiguation from the title string and format the title object similarly to how it is represented in the database. The server code was also modified to accommodate the new format.

## Code Changes
update_topic_titles in server
prepData in TopicEditor.jsx

### CODE REVIEW

File: static/js/TopicEditor.jsx [hunk 1]
```diff
@@ -114,12 +114,52 @@ const TopicEditor = ({origData, onCreateSuccess, close, origWasCat}) => {
...
+                titleObjArray.push(titleObj)
+            });
+        }
+        return titleObjArray
+    };
+
     const prepData = () => {
         // always add category, title, heTitle, altTitles
-        let postData = { category: data.catSlug, title: data.enTitle, heTitle: data.heTitle, altTitles: {}};
+        let postData = { category: data.catSlug, title: data.enTitle, heTitle: data.heTitle, altTitles: {}, titles: []};
```
Comment (nsantacruz) [inline static/js/TopicEditor.jsx:153]: remove `title`, `heTitle` and `altTitles`.

File: static/js/TopicEditor.jsx [hunk 2]
```diff
@@ -114,12 +114,52 @@ const TopicEditor = ({origData, onCreateSuccess, close, origWasCat}) => {
...
         postData.altTitles.en = data.enAltTitles.map(x => x.name); // alt titles implemented using TitleVariants which contains list of objects with 'name' property.
```
Comment (nsantacruz) [inline static/js/TopicEditor.jsx:154]: these lines can be removed and the data can be passed directly to createNonPrimaryTitleObjArray().

File: static/js/TopicEditor.jsx [hunk 3]
```diff
@@ -114,11 +116,43 @@ const TopicEditor = ({origData, onCreateSuccess, close, origWasCat}) => {
...
+
+    const createPrimaryTitleObj = function(rawTitle, lang){
+        let primaryTitleObj = {'text': removeDisambiguationFromTitle(rawTitle), "lang": lang, "primary": true};
+        let disambiguation = extractDisambiguationFromTitle(rawTitle);
+        if (disambiguation) {primaryTitleObj["disambiguation"]=disambiguation};
+        return primaryTitleObj;
+    };
+    const createNonPrimaryTitleObjArray = function(altTitles, lang){
+        const titleObjArray = []
+        if (Array.isArray(altTitles)) {
```
[the remark this answers is not in the document]
Comment (yonadavGit) [reply static/js/TopicEditor.jsx:134]: sometimes (very often) it's null, I can change it to a test the "nullity" of the object in a more straightforward way.

[the remark this answers is not in the document]
Comment (yonadavGit) [reply static/js/TopicEditor.jsx:134]: OK Sorry this check is redundant

File: sefaria/model/tests/topic_test.py [hunk 4]
```diff
@@ -164,6 +165,25 @@ def test_sanitize(self):
         assert "<script>" not in t.description["he"]
         assert "<script>" not in t.slug
 
+    def test_update_topic(self, topic_graph):
```
Comment (nsantacruz) [inline sefaria/model/tests/topic_test.py:168]: move to `helper/tests/topic_test.py`. create a fixture there that creates a single topic for use in the test

**Pull request merged**